src/main/java/core/basesyntax/report/input/FileReaderImpl.java

```diff
@@ -0,0 +1,25 @@
+package core.basesyntax.report.input;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class FileReaderImpl implements Reader {
+    @Override
+    public List<String> read(String srcPath) {
+        try (BufferedReader reader = new BufferedReader(
+                new FileReader(srcPath)
+        )) {
+            List<String> lines = new ArrayList<>();
+            String line;
+            while ((line = reader.readLine()) != null) {
+                lines.add(line);
+            }
+            return lines;
+        } catch (IOException e) {
+            throw new RuntimeException(e);
```

Comment:
According to the checklist, when catching exceptions, it is recommended to rethrow them with an informative message. Consider adding a message to the RuntimeException to inform about the error context, such as the path of the file that failed to be read.